### DISCUSSION TITLE
Question about hash ring in Cortex Architecture

### DISCUSSION THREAD

**Discussion by petersin0422** [cortexproject/cortex#3584] [Q&A] [answered]
> The effect of this hash set up is that each token that an ingester owns is responsible for a range of hashes. If there are three tokens with values 0, 25, and 50, then a hash of 3 would be given to the ingester that owns the token 25; the ingester owning token 25 is responsible for the hash range of 1-25.
>
> - from ["the hash ring"](https://cortexmetrics.io/docs/architecture/#the-hash-ring) section

For the example, where there are 3 ingesters with token 0, 25, 50. If the hashed series from the distributor is non-negative, does that mean we could potentially end up with an ingester that is never being hit? Assuming that the replication factor is 1.

**Comment by gouthamve** [comment, marked as answer]
Yes. But remember that each ingester has 128 tokens and some run with 512 tokens. And beyond that we usually store millions of series, so we find that in practice you'll have quite the optimal distribution.
